package/MDAnalysis/analysis/dssp.py

```diff
@@ -0,0 +1,279 @@
+        unfoldmap_rev[:, :, :, 0, 1] * unfoldmap[:, :, :, 1, 2]
+    )
+    p_bridge = np.pad(p_bridge, [[0, 0], [1, 1], [1, 1]])
+    a_bridge = (unfoldmap[:, :, :, 1, 1] * unfoldmap_rev[:, :, :, 1, 1]) + (
+        unfoldmap[:, :, :, 0, 2] * unfoldmap_rev[:, :, :, 0, 2]
+    )
+    a_bridge = np.pad(a_bridge, [[0, 0], [1, 1], [1, 1]])
+    # ladder
+    ladder = (p_bridge + a_bridge).sum(-1) > 0
+    # H, E, L of C3
+    helix = (helix3 + helix4 + helix5) > 0
+    strand = ladder
+    loop = ~helix * ~strand
+    onehot = np.stack([loop, helix, strand], axis=-1)
+    onehot = rearrange(
+        onehot, "1 ... -> ...") if len(org_shape) == 3 else onehot
+    return onehot
+
+
+def translate(onehot: np.ndarray) -> np.ndarray:
+    """Translate a one-hot encoding ndarray into char-based secondary structure assignment.
+    One-hot encoding corresponds to '-', 'H', 'E' (in that order) -- loop, helix and sheet respectively.
+    Input array must have its last axis of shape 3: (n_residues, 3) or (n_frames, n_residues, 3), etc.
+
+    Example:
+
+    >>> from MDAnalysis.analysis.dssp import translate
+    >>> import numpy as np
+    >>> # encoding 'HE-'
+    >>> onehot = np.array([[False, True, False], [False, False, True], [True, False, False]]])
+    >>> ''.join(translate(onehot))
+    'HE-'
+
+    Args:
+        onehot (np.ndarray): _description_
+
+    Returns:
+        np.ndarray: _description_
+    """
+    C3_ALPHABET = np.array(["-", "H", "E"])
+    index = np.argmax(onehot, axis=-1)
+    return C3_ALPHABET[index]
+
+
+class DSSP(AnalysisBase):
+    """Assign secondary structure using DSSP algorithm as implemented by pydssp package (v. 0.9.0)
+    Here:
+     - 'H' represents a 4-helix (alpha-helix)
+     - 'E' represents 'extended strand', participating in beta-ladder
+     - '-' represents unordered part
+
+    Example:
+    >>> from MDAnalysis.analysis.dssp import DSSP
+    >>> from MDAnalysisTests.datafiles import PDB
+    >>> import MDAnalysis as mda
+    >>> u = mda.Universe(PDB)
+    >>> run = DSSP(u).run()
+    >>> print("".join(run.results.dssp[0]))
+    '--EEEEE-----HHHHHHHHHHHH--EEE-HHHHHHHHHHH--HHHHHHHHHHHH-----HHHHHHHHHHH---HHH---EEEE-----HHHHHHHHHH-----EEEEEE--HHHHHHHH--EE--------EE---E------E------E----HHH-HHHHHHHHHHHHHHHHHHHHHHHHHHHH---EEEEEE----HHHHHHHHHHHH-'
+
+    Also, per-frame dssp assignment allows you to build average secondary structure -- `DSSP.results.dssp_ndarray` 
+    holds (n_frames, n_residues, 3) shape ndarray with one-hot encoding of loop, helix and sheet, respectively:
+
+    >>> from MDAnalysis.analysis.dssp import translate
+    >>> u = mda.Universe(...)
+    >>> long_run = DSSP(u).run()
+    >>> mean_secondary_structure = translate(long_run.results.dssp_ndarray.mean(axis=0))
+    >>> ''.join(mean_secondary_structure)
+    ---HHHHHHHH--------------------------HHHHHH-------HHHH---HHHHHHHHHHHHHH------------HHHHH------------------HHHHHHHH---
```

Comment:
Nice examples. Perhaps a short one in the doc string and extensive ones in the module docs.

Reply:
The module docs now contain 3 big examples, while class doc contains 2 of them -- for trajectory and for single file.